### PR TITLE
feat: add loadNavbarOnCover configuration

### LINKED ISSUE #1375: Ability to disable the showing of the navigation bar on the cover
## Feature request
Ability to disable the showing of the navigation bar on the cover

#### What problem does this feature solve?
This feature request makes it so users have the ability to be able to not show the navigation bar on the cover, and have it only show on the normal pages.

### PR DESCRIPTION
**Pull request by sy-records**
**Summary**

resolve #1375

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Repo settings
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [x] Related documents have been updated
- [x] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

### CODE REVIEW

File: cypress/fixtures/tpl/docs.index.html [hunk 1]
```diff
@@ -109,4 +110,4 @@
   <script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>
 </body>
 
-</html>
\ No newline at end of file
+</html>
```
Comment (anikethsaha) [inline cypress/fixtures/tpl/docs.index.html:113]: can you add Line here

**Review by anikethsaha — CHANGES_REQUESTED** [review]
One change, other than that, LGTM

File: cypress/fixtures/tpl/docs.index.html (hunk 1 above)
Comment (sy-records) [reply cypress/fixtures/tpl/docs.index.html:113]: I don't know how to change it... 
I've tried several times and it still looks like this.

Comment (anikethsaha) [reply cypress/fixtures/tpl/docs.index.html:113]: can you create a new line manually and turn off the auto format on save in your editor.

Comment (sy-records) [reply cypress/fixtures/tpl/docs.index.html:113]: Is it like this now?

Comment (sy-records) [reply cypress/fixtures/tpl/docs.index.html:113]: I think it was the original that didn't have a line feed, that was wrong...
image

Comment (Koooooo-7) [reply cypress/fixtures/tpl/docs.index.html:113]: Interesting. let me have a try.

Comment (Koooooo-7) [reply cypress/fixtures/tpl/docs.index.html:113]: I realize this is fine that it is the origin file lost the new line.

**Review by anikethsaha — APPROVED** [review]

**Review by Koooooo-7 — CHANGES_REQUESTED** [review]
Little problem.
Steps:
1. set ` loadNavbarOnCover: false,`.
2. clink to another selection.
3. clink the `back` button of the browser, it will back to the cover page.
the nav will show on the cover now.

**Comment by sy-records** [comment]
This is caused by not reloading the page, I think that's acceptable.
It just doesn't show up when you visit the home page directly.

**Comment by Koooooo-7** [comment]
> This is caused by not reloading the page, I think that's acceptable.
> It just doesn't show up when you visit the home page directly.

It make no sense, there is no reason to make an element which had been set `not show` visible.
There is one way I found to make it, FYR.
```js
    // Load nav
    if (loadNavbar) {
      // default show the nav
      document.querySelector('.app-nav').style.display = 'inline';
      if (path === '/' && !loadNavbarOnCover) {
        // hidden nav on cover
        document.querySelector('.app-nav').style.display = 'none';
        return;
      }
```
Maybe you could get another brilliant one :).

**Review by Koooooo-7 — COMMENTED** [review]
Well. 
I think the test is not a test actually， it just check the configuration property exist, not checking if it works tho.

Besides that, I think we need concern about refactoring the nav, for some reason. #1230

File: src/core/config.js [hunk 2]
```diff
@@ -35,6 +35,7 @@ export default function(vm) {
       crossOriginLinks: [],
       relativePath: false,
       topMargin: 0,
+      loadNavbarOnCover: true,
```
Comment (anikethsaha) [inline src/core/config.js:38]: how about default `false` ?

WDYT ?

**Review by anikethsaha — CHANGES_REQUESTED** [review]

File: docs/configuration.md [hunk 3]
```diff
@@ -629,3 +629,15 @@ window.$docsify = {
   topMargin: 90, // default: 0
 };
 ```
+
+## loadNavbarOnCover
+
```
Comment (Koooooo-7) [inline docs/configuration.md:634]: How about naming it `hideNavbar`.   😃 and plz checking if there has exceptions when this element removed also.

Comment (sy-records) [reply docs/configuration.md:634]: It’s just whether it’s displayed on the cover, not all pages

File: src/core/config.js (hunk 2 above)
Comment (sy-records) [reply src/core/config.js:38]: We are displaying by default now, so I think it’s better to be consistent.

Comment (anikethsaha) [reply src/core/config.js:38]: makes sense

**Pull request closed without merging**